### DISCUSSION TITLE
AttributeError: 'approve' object has no attribute 'build_transaction'

### DISCUSSION THREAD

**Discussion by eth-orangutan** [uniswap-python/uniswap-python#312] [Q&A]
Hi! Trying to use v3 on Polygon to swap MATIC/USDC. Running into the error in the title (full copy below), Using the latest version of Uniswap Python.

I am able to use get_eth_balance() and get_price_impact just fine, but make_trade runs into the error below.
```

from uniswap import Uniswap
import config
from web3 import Web3
from web3.middleware import geth_poa_middleware

web3 = Web3(Web3.HTTPProvider(config.node_address))
print("Connected to Polygon Node: ", web3.isConnected())

web3.middleware_onion.inject(geth_poa_middleware, layer=0)

address = config.my_wallet_address
private_key = config.my_private_key
version = 3
provider = config.endpoint
uniswap = Uniswap(address=address, private_key=private_key,
                  version=version, web3=web3)


print(uniswap.get_eth_balance())  # works fine

print(uniswap.get_price_input(config.matic_address,
      config.usdc_address, 10**18))  # works fine


uniswap.make_trade(config.matic_address,
                   config.usdc_address, 10**18, fee="500")
```

Traceback (most recent call last):
  File "/home/michael/Documents/2023shit/swap.py", line 27, in <module>
    uniswap.make_trade(config.matic_address,
  File "/home/michael/.local/lib/python3.8/site-packages/uniswap/decorators.py", line 37, in approved
    self.approve(token)
  File "/home/michael/.local/lib/python3.8/site-packages/uniswap/uniswap.py", line 1409, in approve
    tx = self._build_and_send_tx(function)
  File "/home/michael/.local/lib/python3.8/site-packages/uniswap/uniswap.py", line 1443, in _build_and_send_tx
    transaction = function.build_transaction(tx_params)
AttributeError: 'approve' object has no attribute 'build_transaction'

**Comment by matthieuoliveira** [comment]
I'm experiencing this as well. Any progress since January ?
